src/events-handler.ts

```diff
@@ -33,6 +35,7 @@ class EventsHandler {
         context: IContext,
         enabled: boolean,
         featureName: string,
+        impressionData: boolean,
```

Comment:
Would it make sense to use a name like `impressionDataEnabled` instead? It might make the name slightly clearer (seeing as it's a boolean and not _actual_ impression data). Happy to be overruled here, though.

Reply:
jeah I think it would be better, but I just wanted top keep it the same as the featureToggle property, which currently is "impressionData". 

Wdyt?

Reply:
I'm open to either. As long as we've considered the option, I'm happy for you to make a decision on it ☺️  I see how keeping it the same as the toggle property makes sense, but I **personally** think it makes it clearer to use `impressionDataEnabled` or even `impressionDataEnabledForFeature`. It _could_ cause some confusion when handling an impression data event that you also have a subproperty called `impressionData`, so I'm leaning towards being more explicit here.